### PR TITLE
WebGPURenderer: Remove duplication of interleaved attribute buffers.

### PR DESCRIPTION
**Pull request by aardgoose**
Associate attribute GPUBuffers with interleaved attributes' common shared buffer instead of the attribute itself to remove the duplication of GPUBuffers.

Rework vertex buffer layout for the pipeline to remove duplicate buffer entries, and setVertexBuffer() calls to match the new layout.

Rework nodes InstanceNode and BufferAttributeNode to use shared backing buffers for conversion of mat4 to vec4 * 4.
(required to ensure the backing buffer is of the correct type)

### CODE REVIEW

File: examples/jsm/nodes/accessors/BufferAttributeNode.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ class BufferAttributeNode extends InputNode {
 		const stride = this.bufferStride || itemSize;
 		const offset = this.bufferOffset;
 
-		const buffer = new InterleavedBuffer( array, stride );
+		const buffer = ( array.isInterleavedBuffer === undefined ) ? new InterleavedBuffer( array, stride ) : array;
```
Comment (LeviPesin) [inline examples/jsm/nodes/accessors/BufferAttributeNode.js:32]:
```suggestion
		const buffer = array.isInterleavedBuffer === true ? array : new InterleavedBuffer( array, stride );
```

File: examples/jsm/renderers/common/RenderObject.js [hunk 2]
```diff
@@ -69,19 +69,36 @@ export default class RenderObject {
...
 
 		return attributes;
 
 	}
 
+	getVertexBuffers() {
+
+		if ( this.vertexBuffers !== null ) return this.vertexBuffers;
+
+		this.getAttributes();
```
Comment (LeviPesin) [inline examples/jsm/renderers/common/RenderObject.js:96]:
```suggestion
		if ( this.vertexBuffers === null ) this.getAttributes();
```

File: examples/jsm/renderers/webgpu/utils/WebGPUPipelineUtils.js [hunk 3]
```diff
@@ -34,24 +34,10 @@ class WebGPUPipelineUtils {
...
-			vertexBuffers.push( {
-				arrayStride: attribute.arrayStride,
-				attributes: [ { shaderLocation: attribute.slot, offset: attribute.offset, format: attribute.format } ],
-				stepMode: attribute.stepMode
-			} );
-
-		}
-
+		const vertexBuffers = backend.attributeUtils.createShaderVertexBuffers( renderObject );
+console.log( vertexBuffers );
```
Comment (LeviPesin) [inline examples/jsm/renderers/webgpu/utils/WebGPUPipelineUtils.js:40]:
```suggestion

```
Remains from debugging?

Comment (aardgoose) [reply examples/jsm/renderers/webgpu/utils/WebGPUPipelineUtils.js:40]: definitely! Surprised I missed it.

File: examples/jsm/renderers/common/RenderObject.js (hunk 2 above)
Comment (aardgoose) [reply examples/jsm/renderers/common/RenderObject.js:96]: Makes sense. I was echoing the pattern from getAttributes().
I also noticed I missed setting vertexBuffers to null which was in my working copy.

**Pull request merged**